sdk/framework/openapi.go

```diff
@@ -268,34 +268,22 @@ func documentPath(p *Path, backend *Backend, requestResponsePrefix string, doc *
 
 		// Process path and header parameters, which are common to all operations.
 		// Body fields will be added to individual operations.
-		pathFields, bodyFields := splitFields(p.Fields, path)
+		pathFields, queryFields, bodyFields := splitFields(p.Fields, path, captures)
```

Comment:
Note for reviewer: `expandPattern` will now collect extra information about capture groups found in the regex, which gets relayed to `splitFields`. `splitFields` will now make use of that, and cleanly separate path parameters from `Query: true` parameters, which were previously being combined into `pathFields`.